subsys/net/ip/net_core.c

```diff
@@ -392,7 +406,12 @@ int net_send_data(struct net_pkt *pkt)
 static void net_rx(struct net_if *iface, struct net_pkt *pkt)
 {
 #if defined(CONFIG_NET_STATISTICS) || defined(CONFIG_NET_DEBUG_CORE)
-	size_t pkt_len = net_pkt_get_len(pkt);
+	size_t pkt_len;
+#if defined(CONFIG_NET_STATISTICS)
+	pkt_len = pkt->total_pkt_len;
```

Comment:
Ok, I guess you'll need to merge previous patch into this one. Then it's clearer.

Reply:
Sure, makes sense